### DISCUSSION TITLE
Add GoBackTo state transition

### DISCUSSION THREAD

**Discussion by JakBlack** [bazyleu/UniState#90] [Ideas]
Right now, it's hard to separate long flows into states because there's no way to return to core state via history
It's possible to launch these long flows in one state and separate flow into states with separate state machine, but that adds extra comlexity

Add GoBackTo<TSTate> transition, which will find specified state in state history and go back to it, clearing any intermediate state from history
Need to come up with error handling in case specified state isn't found in state history, proposed solution - log an exception, exit the state machine

**Comment by bazyleu** [comment]
Thanks for your suggestion! It is a good idea to add a "GoBackTo" transition to handle long flows easier. If the specified state is not in history, I think it is better to exit the state machine because it is more consistent with classic "GoBack" behavior . We will implement this in future versions.

**Comment by bazyleu** [comment]
For info: GoBackTo was implemented by @JakBlack in version 1.3.0.
